Description: stall some loops in test_gateway a little bit to prevent
 test failure and build break which occurs sometimes on this.
Author: Andreas Cadhalpun <andreas.cadhalpun@googlemail.com>
Bug-Debian: https://bugs.debian.org/840823

--- a/testing/test_gateway.py
+++ b/testing/test_gateway.py
@@ -3,6 +3,7 @@
 mostly functional tests of gateways.
 """
 import os
+import time
 import socket
 import subprocess
 import sys
@@ -104,6 +105,7 @@ class TestBasicGateway:
             status = gw.remote_status()
             if status.numexecuting == 0:
                 break
+            time.sleep(0.1)
         else:
             pytest.fail("did not get correct remote status")
         # race condition
@@ -456,6 +458,7 @@ class TestThreads:
             rstatus = gw.remote_status()
             if rstatus.numexecuting == 0:
                 return
+            time.sleep(0.1)
         assert 0, "numexecuting didn't drop to zero"
 
 
